packages/knip/src/plugins/angular/index.ts

```diff
@@ -23,31 +23,33 @@ const resolveConfig: ResolveConfig<AngularCLIWorkspaceConfiguration> = async (co
 
   for (const project of Object.values(config.projects)) {
     if (!project.architect) return [];
-    for (const target of Object.values(project.architect)) {
+    for (const [targetName, target] of Object.entries(project.architect)) {
+      const isProductionTarget = targetName === 'build';
```

Comment:
I'm totally on board with this. Maybe move this to a function to make it more explicit and then we could pass it e.g. both of `[targetName, target]` to have any sort of heuristics to decide "prod or not".

Reply:
Nice to hear!  Thanks for the early feedback, that's encouraging 💪 ☺️

Eventually implemented the improvement of taking into account the several  `configurations` a target may have. So code has changed and there's now 2 places where the conditions for production are different. Let me know if can be improved with the final implementation.